### PR TITLE
Fix(grid): make trigger menu button accessible

### PR DESCRIPTION
**Pull request by CatalinB7**
The button triggering the menu from the grid's filters was not announcing the current state collapsed / expanded.
Clicking with the mouse (cursor not visible tho, had to pay for that functionality lol):

https://user-images.githubusercontent.com/87014385/223549038-b497a4c6-99e7-4e9d-9450-d0b58152f893.mp4

Navigating using only the keyboard:

https://user-images.githubusercontent.com/87014385/223549066-b9eea743-691e-4571-a4da-74971270d464.mp4

### CODE REVIEW

File: projects/angular/directives/custom-mat-menu-trigger/src/custom-mat-menu-trigger.directive.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+export class UiCustomMatMenuTriggerDirective extends _MatMenuTriggerBase implements AfterContentInit, OnDestroy {
+    @HostBinding()
+    ariaExpanded = false;
+
+    @Input()
+    expandedTranslation = 'expanded';
+
+    get nativeElement() {
+        // eslint-disable-next-line no-underscore-dangle
+        return (this as any)._element.nativeElement as HTMLElement;
```
Comment (anbalase) [inline projects/angular/directives/custom-mat-menu-trigger/src/custom-mat-menu-trigger.directive.ts:29]:
```suggestion
        return inject(ElementRef<HTMLElement>).nativeElement;
```

File: projects/angular/a11y/src/queued-announcer/queued-announcer.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export class QueuedAnnouncer {
     constructor(private _liveAnnouncer: LiveAnnouncer) { }
 
     enqueue(msg: string) {
```
Comment (anbalase) [inline projects/angular/a11y/src/queued-announcer/queued-announcer.ts:19]: let's update the signature as well

File: projects/angular/components/ui-grid/src/ui-grid.intl.ts [hunk 3]
```diff
@@ -118,6 +118,17 @@ export class UiGridIntl implements OnDestroy {
...
+    /**
+     * Menu expanded message for live announcer.
+     *
+     */
+    menuExpanded = 'expanded';
+    /**
+     * Menu collapsed message for live announcer.
+     *
+     */
+    menuCollapsed = 'collapsed';
```
Comment (anbalase) [inline projects/angular/components/ui-grid/src/ui-grid.intl.ts:130]: remove

**Review by anbalase — COMMENTED** [review]
let's add a FIXME for when we upgrade to 15@latest

File: projects/angular/directives/custom-mat-menu-trigger/src/custom-mat-menu-trigger.directive.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    private _liveAnnouncer = inject(LiveAnnouncer);
+    private _destroyed$ = new Subject<void>();
+    ngAfterContentInit() {
+        this.menuOpened.pipe(
+            takeUntil(this._destroyed$),
+        ).subscribe(() => {
+            // although setting aria-expanded to true, it is not announced
+            this.nativeElement.setAttribute('aria-expanded', 'true');
```
Comment (anbalase) [inline projects/angular/directives/custom-mat-menu-trigger/src/custom-mat-menu-trigger.directive.ts:39]: let's pls check if behaviour is consistent across screen readers

Comment (CatalinB7) [reply projects/angular/directives/custom-mat-menu-trigger/src/custom-mat-menu-trigger.directive.ts:40]: I received a confirmation that the vendor uses NVDA, and i tried to install JAWS but it requires elevated permissions

**Comment by caleniuc** [comment]
On mac's VoiceOver, the `expanded` is announced just once. Seems that if you announce the same thing on the `LiveAnnouncer` multiple times, it's announced to the screen reader just the first time. If you call `announce` with a different message, it works.

**Review by caleniuc — APPROVED** [review]

**Pull request merged**